### PR TITLE
[cleanup] factor out get_head_mask, invert_attn_mask, get_extended_attention_mask

### PR DESCRIPTION
**Pull request by sshleifer**
This is three changes applied all over:

1) `get_head_mask` from @LysandreJik is used instead of redundant snippet
2) `get_extended_attention_mask` is used instead of redundant snippet (that also makes causal mask)
3) `invert_attention_mask` is used instead of redundant snippet that doesn't make causal mask.

These changes make the forward passes more readable and allow us to update common logic in one place moving forward! I was reading code last night to try to get a sense of what all the models/tokenizers do and was frustrated with the amount of time spent scrolling through this stuff. Especially for new people, it makes getting to the meet of the `forward` pass much harder to have 100 lines repeated lines of input manipulation at the beginning.

Very open to suggestions.
Open to doing this for TF in a separate PR.
Also if `prune_heads` or other opportunities catch your eye, let me know.

### CODE REVIEW

**Comment by julien-c** [comment]
This looks nice

File: src/transformers/modeling_electra.py [hunk 1]
```diff
@@ -164,65 +164,6 @@ class ElectraPreTrainedModel(BertPreTrainedModel):
     load_tf_weights = load_tf_weights_in_electra
     base_model_prefix = "electra"
 
-    def get_extended_attention_mask(self, attention_mask, input_shape, device):
```
Comment (sshleifer) [inline src/transformers/modeling_electra.py:167]: this is where I got `get_extended_attention_mask` and `get_head_mask`

**Review by thomwolf — APPROVED** [review]
Yes, nice, LGTM

**Review by LysandreJik — COMMENTED** [review]
Love it!
